source/docs/magento2/detect-admin-users.md

```diff
@@ -0,0 +1,178 @@
+
+If you want to customize an existing module instead, you can skip this section.
+
+```js
+// server/modules/admin-data/index.js
+import typeDefs from "./schema.gql";
+import resolvers from "./resolvers.js";
+import { makeUserClientFromRequest } from "server/modules/magento1/core/factories";
+import { AdminDataLoader } from "./loaders";
+
+export default {
+  namespace: "Acme/AdminData",
+  typeDefs: typeDefs,
+  resolvers: resolvers,
+  contextEnhancer: ({ req }) => {
+    const axiosInstance = makeUserClientFromRequest(req);
+
+    return {
+      AdminData: AdminDataLoader(axiosInstance),
+    };
+  },
+};
+```
+
+```js
+// server/modules/admin-data/loaders.js
+import { makeUserClientFromRequest } from "server/modules/magento1/core/factories";
+import { AdminDataLoader } from "./loaders";
+
+export const AdminDataLoader = (axiosInstance) => {
+  return {
+    loadData: async () => {
+      const response = await axiosInstance.get("/admin-custom-endpoint")
+      return response.data
+    },
+  };
+};
+```
+
+```graphql
+# server/modules/admin-data/schema.gql
+extend type Query {
+    adminData: String
+}
+```
+
+```js
+// server/modules/admin-data/resolvers.js
+export default {
+    Query: {
+        adminData: (parent, params, { loaders }) => {
+            return loaders.AdminData.loadData()
+        }
+    }
+}
+```
+
+### Change your loader's request with admin role
+
+First we need to inject the loader that allows to detect the admin role into the loader that will need that information. The loader is `MagentoAdmin` and is made available by the module `Magento2/Admin`. So we must make sure that the dependencies have been updated accordingly.
+
+```diff
+// server/modules/admin-data/index.js
+
+// ...
+
+export default {
+  namespace: "Acme/AdminData",
++  dependencies: [
++    "Magento2/Admin", // ensures that loaders.MagentoAdmin exists
++  ],
+-  contextEnhancer: ({ req }) => {
++  contextEnhancer: ({ req, loaders }) => {
+    const axiosInstance = makeUserClientFromRequest(req);
+
+    return {
+-      AdminData: AdminDataLoader(axiosInstance),
++      AdminData: AdminDataLoader(axiosInstance, loaders.MagentoAdmin),
+    };
+  },
+};
+```
+
+We can then use the `MagentoAdmin` loader to only return data when the admin is connected and use this information to authenticate the request. Please note that in this example, I've added a header using a specific token (`x-custom-header`). But this is an imaginary header for now and depends on how you have implemented `/admin-custom-endpoint` in your Magento. However the goal is to always have a protection on your Magento endpoint to make sure that only a request coming from Front-Commerce and with the correct information will be able to use this endpoint. Otherwise, if your Magento URL is compromised, any user could request sensitive admin data.
```

Comment:
```suggestion
We can then use the `MagentoAdmin` loader to only return data when the admin is connected and use this information to authenticate the request. Please note that this example illustrates how to add a header using a specific token (`x-custom-header`). But this is an imaginary header for now and depends on how you have implemented `/admin-custom-endpoint` in your Magento. However the goal is to always have a protection on your Magento endpoint to make sure that only a request coming from Front-Commerce and with the correct information will be able to use this endpoint. Otherwise, if your Magento URL is compromised, any user could request sensitive admin data.
```